client/src/components/BasicTable.jsx

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react';
+import './table.css';
+import '../App'
+
+
+const RowDetails = () => {
+  return (
+    <p>
+      Some Details
+    </p>
+  )
+}
+
+export const TableRow = (props) => {
+  const [showDetails, setShowDetails] = useState(false);
+
+  const handleRowClick = () => {
+    showDetails ? setShowDetails(!showDetails) : setShowDetails(!showDetails)
+  }
+
+  return (
+    <>
+      <tr onClick={handleRowClick}>
+        <td>{props.product.id}</td>
+        <td>{props.product.name}</td>
+        <td>{props.product.color}</td>
+      </tr>
+      <tr>
+        {showDetails ? RowDetails() : null}
+      </tr>
+    </>
+
+  )
+}
+
+const BasicTable = (props) => {
```

Comment:
Use destructuring, it's makes the code cleaner:
const BasicTable = ({ products }) => {